scapy/layers/dhcp6.py

```diff
@@ -1346,8 +1346,7 @@ def norm_list(val, param_name):
 
         ####
         # Find the source address we will use
-        l = filter(lambda x: x[2] == iface and in6_islladdr(x[0]), 
-                   in6_getifaddr())
+        l = [x for x in in6_getifaddr() if x[2] == iface and in6_islladdr(x[0])]
```

Comment:
Maybe this would be better:
```Python
    try:
        self.src_addr = next(x for x in in6_getifaddr() if x[2] == iface and in6_islladdr(x[0]))[0]
    except StopIteration:
        warning("Unable to get a Link-Local address")
        pass
```

What do you think?